src/Apps/Artist/routes.tsx

```diff
@@ -47,8 +49,20 @@ export const routes = [
       {
         path: "/",
         Component: OverviewRoute,
+        render: ({ props, Component }) => {
+          if (!props) {
+            return null
+          }
+
+          return (
+            <Provider inject={[new FilterState(props.location.query)]}>
+              <Component artist={props.artist} />
+            </Provider>
+          )
+        },
         prepareVariables: (params, props) => {
-          return { ...props.location.query, ...params }
+          const initialFilterState = props.location ? props.location.query : {}
```

Comment:
Can you add a `FIXME` comment that notes the weird behaviour which is probably a bug?